.github/workflows/test.yml

```
@@ -0,0 +1,160 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
```

```suggestion
    paths:
      - "**/*.go"
      - "**/*.c"
      - "**/*.h"
      - "go.mod"
      - "go.sum"
      - ".github/workflows/test.yml"
```

Add trigger conditions (based on files).